MapboxCoreNavigation/RouteController.swift

```diff
@@ -330,6 +339,50 @@ extension RouteController: CLLocationManagerDelegate {
         }
     }
     
+    func checkForFasterRoute(from location: CLLocation) {
+        guard let currentUpcomingManeuver = routeProgress.currentLegProgress.upComingStep else { return }
+        
+        guard let lastLocationDate = lastLocationDate else {
+            self.lastLocationDate = location.timestamp
+            return
+        }
+
+        // Only check ever 2 minutes for faster route
+        guard location.timestamp.timeIntervalSince(lastLocationDate) >= 120 else { return }
+        let durationRemaining = routeProgress.durationRemaining
+        let currentAlertLevel = routeProgress.currentLegProgress.alertUserLevel
+        
+        getDirections(from: location) { [weak self] (route, error) in
+            guard let strongSelf = self else { return }
+            
+            strongSelf.lastLocationDate = nil
+            
+            guard let route = route else { return }
+            
+            if route.legs[0].steps[1].expectedTravelTime <= 70 {
+                let maneuverLocation = CLLocation(coordinate: currentUpcomingManeuver.maneuverLocation, altitude: 0, horizontalAccuracy: 1, verticalAccuracy: 1, course: currentUpcomingManeuver.finalHeading ?? currentUpcomingManeuver.initialHeading ?? location.course, speed: 10, timestamp: Date())
+                
+                strongSelf.getDirections(from: maneuverLocation, completion: { (route, error) in
+                    guard let route = route else { return }
+                    
+                    // Only use new route if it's at least 10% faster
+                    if route.expectedTravelTime <= 0.9 * durationRemaining {
+                        // If the upcoming maneuver in the new route is the same as the current upcoming maneuver, don't announce it
+                        strongSelf.routeProgress = RouteProgress(route: route, legIndex: 0, alertLevel: currentUpcomingManeuver.description == route.legs[0].steps[1].description ? currentAlertLevel : .none)
```

Comment:
`RouteStep.description` is really just an alias for `RouteStep.instructions`, which isn’t guaranteed to be unique among RouteSteps. For example, what if `currentUpcomingManeuver` is a ramp step and the new `route` starts with a different ramp step? In both cases, `instruction` might be:

> Take the ramp on the right

but the new step could be much closer.

For the purposes of this feature, two steps are equivalent if they’re located in the same place and have the same maneuver type and direction. mapbox/MapboxDirections.swift#151 would implement a custom `==` operator for RouteStep, but there’s nothing stopping us from implementing the same logic here for now.